src/agent/Bradley.kt

```diff
@@ -0,0 +1,24 @@
+package agent
+
+import action.ActionControl
+import brain.LearningUnit
+import brain.Memory
+import brain.VisualProcessingUnit
+import map.Node
+
+class Bradley {
+    var lifeUnits = 10
+    var currentNode =  Node(0,0)
```

Comment:
Because node is both <position> and <stuff in that position>, here you actually have not a node object from the Map, but an entirely separate object. This is dangerous and confusing. Breaking Node contents from the position, and/or storing only position in Bradley would fix it.